components/ide-service/pkg/server/server.go

```diff
@@ -97,9 +97,37 @@ func (s *IDEServiceServer) register(grpcServer *grpc.Server) {
 }
 
 func (s *IDEServiceServer) GetConfig(ctx context.Context, req *api.GetConfigRequest) (*api.GetConfigResponse, error) {
-	return &api.GetConfigResponse{
-		Content: s.parsedIDEConfigContent,
-	}, nil
+	configCatClient := experiments.NewClient()
+	attributes := experiments.Attributes{
+		UserID:    req.User.Id,
+		UserEmail: req.User.GetEmail(),
+	}
+
+	experimentalIdesEnabled := configCatClient.GetBoolValue(ctx, "experimentalIdes", false, attributes)
+
+	if experimentalIdesEnabled {
+		// We can return everything
+		return &api.GetConfigResponse{
+			Content: s.parsedIDEConfigContent,
+		}, nil
+	} else {
+		for key, ide := range s.ideConfig.IdeOptions.Options {
+			if ide.Experimental && !experimentalIdesEnabled {
+				delete(s.ideConfig.IdeOptions.Options, key)
+			}
+		}
+
+		parsedConfig, err := json.Marshal(s.ideConfig)
+		if err != nil {
+			log.WithError(err).Error("cannot marshal ide config")
+			return nil, err
+		}
+		s.parsedIDEConfigContent = string(parsedConfig)
```

Comment:
We should not update like that. It is shared value for all users.

Reply:
@akosyakov made changes in https://github.com/gitpod-io/gitpod/pull/17196/commits/12c1b0df83fff88d15abed4c2f8ae9560cb772ce, looks better?

Reply:
yes, not sure whether we need to preserve other properties of IDE config, both branches should return similar shapes to avoid surprises later that one return only one property but other all

Reply:
Why don't precompute 2 properties instead and then return either one or another?

Reply:
Nice point for optimization! Let's do that

Reply:
fc096e4d96419962c3def8b5de6f35920f39e9f3

Reply:
looks good, but you need to test :)